### DISCUSSION TITLE
open dicom folder with Weasis Windows right mouse button menu

### DISCUSSION THREAD

**Discussion by nhaliasos** [nroduit/Weasis#535] [General]
as per my subject of the thread, is there any way i can integrate Weasis to open a dicom folder or even a zip folder with a dicom directory in it by right click on the folder and have an entry to open with Weasis? in Windows

**Comment by nroduit** [comment]
To my knowledge, this concept does not exist in Windows. An application cannot be associated only with a file.

Directories can be opened with [the dicom:get command](https://weasis.org/en/basics/commands/index.html#dicomget).

**Reply by nhaliasos** [reply]
For example, when Radiant installs, an entry appears on the windows registry where you can right mouse click on a dicom folder and "Open with Radiant Dicom viewer" appears as an option amongst the rest of the menu options. A similar scenario is described here: https://superuser.com/questions/1303165/editing-the-open-with-menu-in-windows-10
The question is whether we could do the same for a whole folder to be imported and viewed by right click option?

**Reply by macgarnigle** [reply]
I was able to do this at the folder level with a registry edit. 
Computer\HKEY_CLASSES_ROOT\Folder\shell\openwweasis\command
"C:\Program Files\Weasis\Weasis.exe" dicom:get -l "%1"

**Comment by nroduit** [comment]
The reference you give is the one already implemented in Weasis for associating a file type.
![image](https://github.com/nroduit/Weasis/assets/993975/4064df7f-c9d3-416d-8587-3ca21ab441cc)

Association for a directory doesn't seem to be standard. Some applications have it, but only on Windows 10 and not on Windows 11.

In Weasis, we don't want to implement system-specific features.
